paddle/majel/util.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#define STRINGIFY(x) #x
+#define TOSTRING(x) STRINGIFY(x)
+
+#if defined(__APPLE__) && defined(__CUDA_ARCH__) && !defined(NDEBUG)
+#include <stdio.h>
+#define MAJEL_ASSERT(e)                                                       \
+  do {                                                                        \
+    if (!(e)) {                                                               \
+      printf(                                                                 \
+          "%s:%d Assertion `%s` failed.\n", __FILE__, __LINE__, TOSTRING(e)); \
+      asm("trap;");                                                           \
+    }                                                                         \
+  } while (0)
+
+#define MAJEL_ASSERT_MSG(e, m)                      \
+  do {                                              \
+    if (!(e)) {                                     \
+      printf("%s:%d Assertion `%s` failed (%s).\n", \
+             __FILE__,                              \
+             __LINE__,                              \
+             TOSTRING(e),                           \
+             m);                                    \
+      asm("trap;");                                 \
+    }                                               \
+  } while (0)
+#else
+#include <assert.h>
+#define MAJEL_ASSERT(e) assert(e)
+#define MAJEL_ASSERT_MSG(e, m) assert((e) && (m))
+#endif
+
+namespace majel {
+namespace detail {
+
+inline int div_up(int x, int y) { return (x + y - 1) / y; }
```

Comment:
This function `div_up` is not used in the ported part of Majel. Also, I checked that it's used in the following files of Majel:

```
$ for i in $(du -a | grep '\.h$' | cut -f 2); do if grep 'div_up' $i > /dev/null; then echo $i; fi; done
./detail/bsxfun.h
./detail/util.h
./gpu/detail/gather.h
./gpu/detail/gather_with_indices.h
./gpu/detail/ops.h
./gpu/detail/quantize_ops.h
./gpu/detail/reduce.h
./gpu/detail/row_conv_ops.h
./gpu/detail/scatter_with_indices.h
```

I am not sure if we will need it soon. How about we don't port it right now?  If so, we can name this header file `paddle/majel/detial/assert`.